File: 이지수/Python/Baekjoon_1013.py
def solution(string) :
    while len(string) :
        num = 0
        if string[0] == "0" :
            if len(string) < 2 or string[0:2] != "01": return False
            string = string[2:]
        else :
            if len(string) < 4 or string[0:3] != "100" : return False
            string = string[3:]
            while string[0] == "0":
                if len(string) == 1 : return False
                string = string[1:]
            while string[0] == "1":
                num += 1
                if len(string) > 1: string = string[1:]
                else : string = ""; break
            if num > 1 and string[0:2] == "00" : string = "1" + string
    return True

t = int(input())
for _ in range(t) :
    string = input()
    print("YES" if solution(string) else "NO")

File: 이지수/Swift/Baekjoon_1013.swift
let t = Int(readLine()!) ?? 0
for _ in 0..<t { print(solution(Array(readLine()!)) ? "YES" : "NO")}

@inline(__always)
func solution(_ str: [Character]) -> Bool {
    var str = str
    while !str.isEmpty {
        switch str.first {
        case "0":
            if str.count < 2 || str[1] != "1" { return false }
            str.removeFirst(2)
        case "1":
            if str.count < 4 || str[1] != "0" || str[2] != "0" { return false }
            str.removeFirst(2)
            str.removeFirst(str.firstIndex() {$0 != "0"} ?? str.count)
            if str.first != "1" { return false }
            let index = str.firstIndex() {$0 != "1"} ?? str.count
            str.removeFirst(index)
            if index > 1 && str.count > 2 && str[0] == "0" && str[1] == "0" { str.insert("1", at: 0) }
        default: return false
        }
    }
    return true
}
